jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ApplyService.java

```diff
@@ -1390,4 +1398,55 @@ public void processCustomEntities(List<String> customResourceDefinitions, File r
             }
         }
     }
+
+    public static void applyNamespaceOrProjectIfPresent(ApplyService applyService, Collection<HasMetadata> entities) {
+        List<HasMetadata> projectOrNamespaceList = entities.stream()
+                .filter(e -> e instanceof Namespace || e instanceof Project)
+                .collect(Collectors.toList());
+
+        for (HasMetadata projectOrNamespace : projectOrNamespaceList) {
+            if (projectOrNamespace instanceof Namespace) {
+                Namespace ns = (Namespace) projectOrNamespace;
+                applyService.applyNamespace(ns);
+            } else if (projectOrNamespace instanceof Project) {
+                Project project = (Project) projectOrNamespace;
+                applyService.applyProject(project);
+            }
+        }
+    }
+
+    public static void applyEntities(ApplyService applyService, String fileName, Set<HasMetadata> entities, KitLogger serviceLogger,
+                                 long serviceUrlWaitTimeSeconds, ResourceConfig resources, File resourceDir,
+                                 String environment) throws Exception {
+        // Apply all items
+        applyStandardEntities(applyService, fileName, entities);
+        logExposeServiceUrl(applyService, entities, serviceLogger, serviceUrlWaitTimeSeconds);
+        applyService.processCustomEntities(resources != null ? resources.getCustomResourceDefinitions() : null,
+                resourceDir, environment, resources != null ? resources.getRemotes() : null);
+    }
+
+    private static void logExposeServiceUrl(ApplyService applyService, Set<HasMetadata> entities, KitLogger serviceLogger, long serviceUrlWaitTimeSeconds) throws InterruptedException {
```

Comment:
This has no sense.
You have a private static method where you provide an instance of the containing class? Just remove the static modifier then.